quinn-proto/src/connection.rs

```diff
@@ -3378,3 +3475,36 @@ const ACK_DELAY_EXPONENT: u8 = 3;
 const MAX_BACKOFF_EXPONENT: u32 = 16;
 // Minimal remaining size to allow packet coalescing
 const MIN_PACKET_SPACE: usize = 40;
+
+/// Errors that can arise when sending a datagram
+#[derive(Debug, Error, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum SendDatagramError {
+    /// The connection does not have capacity for an additional datagram at this time
+    #[error(display = "sending blocked")]
+    Blocked,
+    /// The datagram is larger than the connection can currently accommodate
+    ///
+    /// Indicates that the path MTU minus overhead or the limit advertised by the peer has been
+    /// exceeded.
+    #[error(display = "too large")]
+    TooLarge,
+    /// The peer does not support receiving datagram frames
+    #[error(display = "datagrams not supported by peer")]
+    UnsupportedByPeer,
+    /// Datagram support is disabled locally
+    #[error(display = "datagram support disabled")]
+    Disabled,
+}
+
+/// The largest unreliable datagram size that can be transmitted to a willing peer over any path
```

Comment:
I found the comment here quite confusing. The comment starts with "the largest" while the constant is named `MIN_`, so I'm already scratching my head. The `_MTU` suffix here seems to suggest that we're dealing with MTUs but I think this actually something like the payload size for a datagram frame that just fits in the QUIC minimum MTU? In any case it is not the minimum, because that is imposed by the transport parameter.